modular_zubbers/code/modules/storyteller/event_defines/roleset/_antagonist_event.dm

```diff
@@ -83,8 +85,20 @@
 	typepath = /datum/round_event/antagonist/solo
 
 /datum/round_event_control/antagonist/proc/get_antag_amount()
+
 	var/people = SSgamemode.get_correct_popcount()
 	var/amount = base_antags + FLOOR(people / denominator, 1)
+
+	if(antag_datum && maximum_antags_global > 0)
+		var/antag_slots_left = maximum_antags_global
+		for(var/datum/antagonist/existing_antagonist as anything in GLOB.antagonists)
+			if(!istype(existing_antagonist,antag_datum)) //Obviously ignore other antagonists.
+				continue
+			antag_slots_left-- //Slot is occupied.
+			if(antag_slots_left <= 0) //No point in checking anymore.
+				break
```

Comment:
Might be nice to check `isnull(existing_antagonist.owner.current)` and exclude those. I do wanna hear your input on this though

Reply:
If it is possible for antag datums not to have owners, then sure. I don't know how they're handled.

Reply:
If the `owner.current` is missing then it means the body is destroyed so it shouldn't be taking up a slot. Also apparently sometimes antag datums can become completely ownerless so `owner` should be nullchecked too